cluster/gce/gci/configure.sh

```diff
@@ -160,8 +160,15 @@ function validate-hash {
 # Get default service account credentials of the VM.
 GCE_METADATA_INTERNAL="http://metadata.google.internal/computeMetadata/v1/instance"
 function get-credentials {
+  local status=""
   curl --fail --retry 5 --retry-delay 3 --retry-connrefused --silent --show-error "${GCE_METADATA_INTERNAL}/service-accounts/default/token" -H "Metadata-Flavor: Google" -s | ${PYTHON} -c \
     'import sys; import json; print(json.loads(sys.stdin.read())["access_token"])'
+  status="$?"
+
+  if [[ "${status}" -ne 0 ]]; then
+    echo "Cannot download a default service account token for this VM." >&2
+    exit ${status}
```

Comment:
not sure on error semantics here, but maybe it makes sense to simply return the status code here as before and instead perform this check during the call of `get-credentials`.

https://github.com/kubernetes/kubernetes/blob/d3dec1602deb9d9607815a20f3f1b0f9f55671f9/cluster/gce/gci/configure.sh#L192-L193
 
The reason I bring that up is AFAIU the semantics here are a bit confusing currently. `get-credentials` is called within a subshell so the `exit ${status}` doesn't actually exit the main script (rather only the subshell). However what is happening as I understand is during error, this would return a failure exit code and then during the `$(get-credentials)` call in download-or-bust the script will exit due to failure exit code and `configure.sh` run under `set -o errexit`.

So I think it would more straightforward if the error check was done in the caller `download-or-bust` instead, i.e leave this function unchanged and in download-or-bust change it to something like:

```
if ! bearer_token=$(get-credentials); then
  echo "Cannot download a default service account token for this VM."
  exit $?
fi
curl_headers="Authorization: Bearer ${bearer_token}"
```

Reply:
changed to caller-initiated error check